airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination_async/DetectStreamToFlush.java

```diff
@@ -127,9 +129,11 @@ Optional<StreamDescriptor> getNextStreamToFlush(final long queueSizeThresholdByt
    */
   @VisibleForTesting
   ImmutablePair<Boolean, String> isTimeTriggered(final StreamDescriptor stream) {
-    final Boolean isTimeTriggered = bufferDequeue.getTimeOfLastRecord(stream)
-        .map(time -> time.isBefore(Instant.now().minus(MAX_TIME_BETWEEN_REC_MIN, ChronoUnit.MINUTES)))
-        .orElse(false);
+    final Boolean isTimeTriggered = lastTimeCalled <= (System.currentTimeMillis() - 5 * 60 * 1000);
+
+    if (isTimeTriggered) {
+      lastTimeCalled = System.currentTimeMillis();
+    }
```

Comment:
Why replace the debounce trigger logic instead of creating another trigger?

Reply:
Also, the lastTimeCalled needs to be updated after a flush for either trigger type and should probably be threadsafe/atomic

Reply:
Done, I move it to a map in order to have a similar logic of flush per stream than what was set before.